plugins/xmlimport/todo.txt

```diff
@@ -0,0 +1,12 @@
+support loading of extensions
+loading of values:
+    I think here we would need more infos about a struct?
+rethink char* handling: hashtable to reduce duplicate strings
+loading of datatypes:
+    nodes are loaded, but the I think it don't makes sense to generate the Types_Array.
+    How would you then instantiate for example a struct? There's no c type generated for it?
```

Comment:
When you are programming a generic server application, you can read/write these structs by direc memory operations (set fields at `void * + Offset`) and work inside the application with the names of the members (which can be read from the XML).
But this is definitely an advanced topic, which in my view does not need to be dealt with during the initial implementation.

Reply:
Thank you for your comment. Did I understand that right that you mean that loading values (also user defined struct values) should not be part of an initial implementation?

Nevertheless, for loading the value into a variable node like this https://github.com/open62541/open62541/blob/6427010002a44a927dd2ba22b51acf8d31659329/examples/nodeset/testnodeset.xml#L264-L281

the whole information about the struct type will be needed (memSize, padding...). So I would also build up a DataTypeArray, consisting the information about the datatypes in the file, add it to the custom types and then set the variable values with this information available.

Reply:
I'm not a Maintainer, so skipping the loading of structured valued is my personal opinion. If you are already half way done, you can also open an additional pull request with this part.

Parsing structured Datatypes:
When a structured datatype is already known in the Server, you can get all the requrired Information from the UA_TYPES (or the user-Version of it) . Each Type is described by this defintion:
https://github.com/open62541/open62541/blob/6427010002a44a927dd2ba22b51acf8d31659329/include/open62541/types.h#L854-870
By utilizing this structure, you can map the name (from the XML) to size, and padding and all the required informations (if type names are enabled).